pkg/k8sutils/resources.go

```diff
@@ -17,40 +17,15 @@ limitations under the License.
 package k8sutils
 
 import (
-	"context"
 	"fmt"
-
-	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/types"
-	"sigs.k8s.io/controller-runtime/pkg/client"
+	"io/ioutil"
 )
 
-// GetTokenFromSA gets the token associated to the first secret located in a k8s' service account
-func GetTokenFromSA(cli client.Client, ns, saName string) (string, error) {
```

Comment:
I might be completely misunderstanding this ... but with this change it seems to me that now we are just reading the secret for the SA the `vault-secrets-manager` pod is running with instead of the one defined in the CR

Reply:
Yes, you are right! I missed this part :( 

We deploy the `vault-secrets-manager` to every namespace (with a single value for `WATCH_NAMESPACE` env) and then using the SA of the operator makes more sense... However, indeed, this kills the possibility of using single-operator installation and keeping more granular access to Vault.  I guess, the PR will be declined

Reply:
Yes security wise, it's bad. The operator should connect on behalf of the client SA.

I currently have very little access to internet and only have my phone, so I can't really work on that before Sept. I am afraid... I can merge PR though ;)

Reply:
Merging this PR will break installations, which watch multiple namespaces. I'm converting the PR into a draft